StRoot/StFstHitMaker/StFstHitMaker.cxx

```diff
@@ -0,0 +1,208 @@
+
+			for (std::vector< StFstCluster * >::iterator clusterIter = clusterCollectionPtr->getClusterVec().begin(); clusterIter != clusterCollectionPtr->getClusterVec().end(); ++clusterIter) {
+				idTruth         = (*clusterIter)->getIdTruth();
+				key             = (*clusterIter)->getKey();
+				disk            = (*clusterIter)->getDisk();
+				wedge           = (*clusterIter)->getWedge();
+				sensor          = (*clusterIter)->getSensor();
+				apv             = (*clusterIter)->getApv();
+				meanRStrip      = (*clusterIter)->getMeanRStrip();
+				meanPhiStrip    = (*clusterIter)->getMeanPhiStrip();
+				maxTb           = (*clusterIter)->getMaxTimeBin();
+				charge          = (*clusterIter)->getTotCharge();
+				chargeErr       = (*clusterIter)->getTotChargeErr();
+				nRawHits        = (*clusterIter)->getNRawHits();
+				nRawHitsR       = (*clusterIter)->getNRawHitsR();
+				nRawHitsPhi     = (*clusterIter)->getNRawHitsPhi();
+				nClusteringType = (*clusterIter)->getClusteringType();
+
+				StFstHit *newHit = new StFstHit(disk, wedge, sensor, apv, charge, chargeErr, maxTb, meanRStrip, meanPhiStrip, nRawHits, nRawHitsR, nRawHitsPhi);
+				newHit->setId(key);
+				newHit->setIdTruth(idTruth);
+
+                                int moduleIdx;
+				if(disk == 1) // Disk 1
+					moduleIdx = wedge;
+				else if(disk == 2)// Disk 2
+					moduleIdx = wedge-12;
+				else if(disk == 3)// Disk 3
+					moduleIdx = wedge-24;
+				// The simple transformation will be updated with the geomtry table in database later
+				if(disk == 1 || disk == 3)
+				{// Disk 1 & 3
+					phiInner = kFstphiStart[moduleIdx-1]*TMath::Pi()/6.0 + 0.5*kFstzDirct[moduleIdx-1]*kFstStripPitchPhi;
+					phiOuter = kFstphiStop[moduleIdx-1]*TMath::Pi()/6.0  - 0.5*kFstzDirct[moduleIdx-1]*kFstStripPitchPhi;
+				}
+				else if(disk == 2)
+				{ // Disk 2
+					phiInner = kFstphiStop[moduleIdx-1]*TMath::Pi()/6.0  - 0.5*kFstzDirct[moduleIdx-1]*kFstStripPitchPhi;
+					phiOuter = kFstphiStart[moduleIdx-1]*TMath::Pi()/6.0 + 0.5*kFstzDirct[moduleIdx-1]*kFstStripPitchPhi;
+				}
+				double local[3];
+				if(meanRStrip < kFstNumRStripsPerWedge/2)
+				{ // inner
+					local[0] = kFstrStart[meanRStrip] + 0.5*kFstStripPitchR;
+					local[1] = phiInner + kFstzFilp[disk-1]*kFstzDirct[moduleIdx-1]*meanPhiStrip*kFstStripPitchPhi;
+				}
+				else
+				{// outer
+					if(sensor == 1){
+						local[0] = kFstrStart[meanRStrip] + 0.5*kFstStripPitchR;
+						local[1] = phiOuter - kFstzFilp[disk-1]*kFstzDirct[moduleIdx-1]*meanPhiStrip*kFstStripPitchPhi - kFstzFilp[disk-1]*kFstzDirct[moduleIdx-1]*0.5*kFstStripGapPhi;
+					}
+					if(sensor == 2){
+						local[0] = kFstrStart[meanRStrip] + 0.5*kFstStripPitchR;
+						local[1] = phiOuter - kFstzFilp[disk-1]*kFstzDirct[moduleIdx-1]*meanPhiStrip*kFstStripPitchPhi + kFstzFilp[disk-1]*kFstzDirct[moduleIdx-1]*0.5*kFstStripGapPhi;
+					}
+				}
+
+				if(disk == 1) local[2] = 146.6012; //unit: cm
+				else if(disk == 2) local[2] = 160.1724; //unit: cm
```

Comment:
please change to (1) 151.750  (2) 165.248  (3) 178.781  from survey document as we agreed upon Monday. With that code is fine

Reply:
update the number in the new commit